# Exploration: broken date tags

#### Date-tag resolution definitions
Based on whether a regex pattern matched on a date-tag
```
year : \d{4}-xx-xx
month : \d{4}-\d{2}-xx
day : \d{4}-\d{2}-\d{2}
broken : none of the above 
```

In [1]:
import re
import ndjson
import pandas as pd

import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
# import primitives
path_primitives = '../data/primitives_220303/primitives.ndjson'
with open(path_primitives) as fin:
    primitives = ndjson.load(fin)

# import date tags
path_dates = '../data/primitives_220303/df_date.csv'
df_date = pd.read_csv(path_dates)

#### overview of corpus by date tag resolution

In [3]:
df_date.groupby(['resolution']).describe()

call_nr                                    date                     \
             count unique                 top  freq  count unique         top   
resolution                                                                      
broken         557     40   1668_Gent_Bill_07   164    557    446  xxxx-xx-xx   
day          15974     63      1574_Antw_Haec  1357  15974   8951  1776-09-14   
month         1450     61      1574_Antw_Haec   256   1450    755  1566-04-xx   
year          6385     62  1666_Gent_Bill_01B   899   6385   1238  1584-xx-xx   

                 
           freq  
resolution       
broken       25  
day         110  
month        16  
year        184

#### where are broken tags?

In [4]:
(df_date
    .groupby(['resolution', 'call_nr'])
    .describe()
    .query('resolution == "broken"')
    .sort_values(by=[('date', 'count')], ascending=False)
)

date                             
                             count unique              top freq
resolution call_nr                                             
broken     1668_Gent_Bill_07   164    124             1003    4
           1602_Brus_Pott      111    109         jaer LXX    2
           1575_Antw_Ulle       54     46       x524-xx-xx    2
           1791_Purm_Louw_01    36     22       xxxx-xx-xx   14
           1574_Antw_Haec       26     23       xxxx-04-01    3
           1628_Alkm_Wijn       21     15       145x-xx-xx    4
           1719_Hoor_Spru       17      4       xxxx-xx-xx    9
           1791_Purm_Louw_02    14     10       1766-03-x8    3
           1687_Rott_Waer       11      9       xxxx-03-02    2
           1671_Rott_Lois       10     10            1347.    1
           1571_Gent_Camp        9      9   sente Sijmoen     1
           1680_Mech_Vale        8      8       09xx-xx-xx    1
           1595_Iepe_Hern_01     8      7       xxxx-06-20    2
           1716_Jisp_Anon        7      5       1666-xx-20    2
           1587_Iepe_Anon        7      6       xxxx-07-28    2
           1576_Zier_Clae        7      6       157x-06-24    2
           1609_Oist_Hoev        5      5            1609.    1
           1604_Bosc_Anon        4      3       xxxx-03-07    2
           1570_Rott_Alle        3      3       1398-xx-29    1
           1668_Gent_Bill_03     3      2       1662-xx-14    2
           1791_Purm_Louw_03     3      3       177x-xx-xx    1
           1574_Antw_EykP        3      3       1566-xx-18    1
           1791_Purm_Louw_04     2      2       1787-06-x0    1
           1674_Utre_Boot        2      2       1672-07-2x    1
           1791_Purm_Louw_05     2      2  Primo Maij 1790    1
           1668_Gent_Bill_12     2      1       1667-xx-07    2
           1648_Rott_Waer        2      2       xxxx-12-xx    1
           1646_Gron_Gijs        2      2       1623-xx-05    1
           1643_Antw_Anon        2      2            1542-    1
           1576_Utre_Anon        2      2       xxxx-06-23    1
           1671_Mede_Oppe        1      1       16xx-02-xx    1
           1675_Brug_Cler        1      1       1529-xz-xx    1
           1668_Gent_Bill_10     1      1       xxxx-xx-26    1
           1681_Zwol_Catt        1      1       xxxx-05-22    1
           1668_Gent_Bill_04     1      1       xxxx-xx-xx    1
           1713_Hout_Dres        1      1             1676    1
           1735_Dokk_Hesm        1      1         1694-04-    1
           1592_Cule_Vrem        1      1        1570-10-8    1
           1581_Haar_Verw        1      1       158x-10-04    1
           1668_Gent_Bill_05     1      1             1664    1

#### broken chronicle 1: gent police
problem: non-standardized annotation (e.g. 1003 instead of 1003-xx-xx )

In [5]:
(df_date
    .query('call_nr == "1668_Gent_Bill_07"')
    .query('resolution == "broken"')
    .groupby('date')
    .size()
    .to_frame(name='count')
    .sort_values(by='count', ascending=False)
)

count
date            
1003           4
1067           3
941            3
984            3
985            3
...          ...
1120           1
1119           1
1118           1
1114           1
Jaer 1664      1

[124 rows x 1 columns]

#### broken chronicle 2: brussels
tagged, but empty attributes (e.g. jaer LXX)

In [6]:
(df_date
    .query('call_nr == "1602_Brus_Pott"')
    .query('resolution == "broken"')
    .groupby('date')
    .size()
    .to_frame(name='count')
    .sort_values(by='count', ascending=False)
)

count
date                                                 
Assensiondach                                       2
jaer LXX                                            2
iersten januario op\t\t\t een jaerdach, ao LII      1
heylich sacramentdach                               1
dit jaer van LXXXII                                 1
...                                               ...
VIIIsten octobre                                    1
VIIIsten februario ao LXXI                          1
Sint Andries-dach                                   1
Sinssendach ende den twedren dach                   1
xx79-xx-xx                                          1

[109 rows x 1 columns]